lib/typed/serializer.rb

```diff
@@ -28,6 +28,10 @@ def deserialize(source)
     def serialize(struct)
     end
 
+    sig { abstract.params(struct: T::Struct).returns(Output) }
+    def deep_serialize(struct)
+    end
```

Comment:
thought: I don't love adding a new method that all Serializers must implement. I don't see how the behavior  would differ on any other serializer except for hashes. Further, I'd also expect a corollary method, a `deep_deserialize`, and that seems weird as well.

This feels like more of an option (passed into the initializer) for the HashSerializer's current serialize method. We could also accept it in the `to_hash` method on `struct_ext`. What do you think?